src/Interceptor/DecompressResponse.php

```diff
@@ -29,30 +29,20 @@ public function requestViaNetwork(
         Cancellation $cancellation,
         Stream $stream
     ): Response {
-        // If a header is manually set, we won't interfere
```

Comment:
See here, there's even a comment explaining that it will do nothing if the header is manually set.